### DISCUSSION TITLE
Source code for demos (northwind)

### DISCUSSION THREAD

**Discussion by chekyassine** [serenity-is/Serenity#5963] [Help]
Hi everyone, 
In new releases of serenity, demos are imported as packages, thus source code not accessible and editable. That was a good feature since one could learn and modify serenity demo source code.
 
I tried to get it back by deleting the packages and adding source code from common-features repository, but lots of errors show. 
Has anyone succeeded in importing demos source code ? 

Thanks !

**Comment by VictorTomaili** [comment]
clone this repository to see the source code of samples 
https://github.com/serenity-is/Serene

**Reply by chekyassine** [reply]
@VictorTomaili , thank you for this suggestion, but sadly, i wasn't able to make it work. 
![image](https://user-images.githubusercontent.com/5337918/132999351-3c25a1c8-5863-4018-a28c-a505644befb4.png)
Furthermore, what I found very useful in previous versions was to simply create a new project using vs template (vsix) and find northwind sourcecode available to modify and test and see impact on the application. Nothing better than this to learn the framework. 

Hopefully, i'll find a way to get that working again.

**Reply by chekyassine** [reply]
I think this is something to be included in the next release. Why not add an option when creating a new project to select if you want demo sources loaded.
Otherwise Serenity.is/demo is enough if what you need is to play with demos.

**Reply by VictorTomaili** [reply]
Ensure about git submodules are initialized and updated.

**Reply by chekyassine** [reply]
Project build takes infinitely long and gives errors 
![image](https://user-images.githubusercontent.com/5337918/133003198-48b01ed8-e523-4cfd-bf27-2f4a2b30656c.png)
I think a newbie should be able to play with demo source code without having to build the huge entire serenity solution including core , assets, service, extension etc etc... 
I believe serenity demo sources should be available in a new project, otherwise what's the value in adding demos in a new project ?

**Reply by chekyassine** [reply]
Hi, to anyone facing same issue , I managed to get it working by deleting the demo packages referenced and adding code from "common-features " repository after changing base package from Serenity.Demos.Northwind to AnythingYouLike.Northwind.
After some trial and error, I eventually got it working, wasn't fun though !
To Serenity Gurus, please get this back, I personally see no point in having demos in a new project without having sources, that's the same as browsing serenity.is/demo.
